File: HRForz-API/app/core/security/password.py
import re

import bcrypt

PASSWORD_PATTERN = re.compile(
    r"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&#^()_+\-=\[\]{}|;:',.<>?/`~])[A-Za-z\d@$!%*?&#^()_+\-=\[\]{}|;:',.<>?/`~]{8,}$"
)


def hash_password(plain: str) -> str:
    return bcrypt.hashpw(plain.encode(), bcrypt.gensalt()).decode()


def verify_password(plain: str, hashed: str) -> bool:
    return bcrypt.checkpw(plain.encode(), hashed.encode())


def validate_password_strength(password: str) -> bool:
    return bool(PASSWORD_PATTERN.match(password))

File: HRForz-API-Node/src/core/security/password.ts
import bcrypt from 'bcryptjs';

const PASSWORD_PATTERN =
  /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&#^()_+\-=\[\]{}|;:',.<>?/`~])[A-Za-z\d@$!%*?&#^()_+\-=\[\]{}|;:',.<>?/`~]{8,}$/;

export async function hashPassword(plain: string): Promise<string> {
  const salt = await bcrypt.genSalt(10);
  return bcrypt.hash(plain, salt);
}

export async function verifyPassword(plain: string, hashed: string): Promise<boolean> {
  return bcrypt.compare(plain, hashed);
}

export function validatePasswordStrength(password: string): boolean {
  return PASSWORD_PATTERN.test(password);
}
